lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -981,6 +981,340 @@ int parse_ncellmeas(const char *at_response, struct lte_lc_cells_info *cells)
+		goto clean_exit;
+	}
+
+	err = at_parser_params_from_str(at_response,
+					NULL,
+					&resp_list);
+	if (err && err != -E2BIG) {
+		LOG_ERR("Could not parse AT%%NCELLMEAS response, error: %d", err);
+		goto clean_exit;
+	} else if (err == -E2BIG) {
+		incomplete = true;
+		LOG_WRN("E2BIG was returned, continue. param_count %d, err %d, str %s",
+			param_count, err, at_response);
+	}
+
+	err = at_params_string_get(&resp_list,
+				   AT_RESPONSE_PREFIX_INDEX,
+				   response_prefix,
+				   &response_prefix_len);
+	if (err) {
+		LOG_ERR("Could not get response prefix, error: %d", err);
+		goto clean_exit;
+	}
+
+	if (!response_is_valid(response_prefix, response_prefix_len,
+			       AT_NCELLMEAS_RESPONSE_PREFIX)) {
+		/* The unsolicited response is not a NCELLMEAS response, ignore it. */
+		LOG_ERR("Not a valid NCELLMEAS response");
+		goto clean_exit;
+	}
+
+	/* Status code. */
+	curr_index = AT_NCELLMEAS_STATUS_INDEX;
+	err = at_params_int_get(&resp_list, curr_index, &status);
+	if (err) {
+		LOG_DBG("Cannot parse NCELLMEAS status");
+		goto clean_exit;
+	}
+
+	if (status != AT_NCELLMEAS_STATUS_VALUE_SUCCESS) {
+		err = 1;
+		LOG_DBG("NCELLMEAS status %d", status);
+		goto clean_exit;
+	}
+
+	/* Go through the cells. */
+	for (i = 0; curr_index < (param_count - (AT_NCELLMEAS_GCI_CELL_PARAMS_COUNT + 1)) &&
+			i < params->gci_count; i++) {
+		struct lte_lc_cell parsed_cell;
+		bool is_serving_cell;
+		uint8_t parsed_ncells_count;
+
+		/* <cell_id>  */
+		curr_index++;
+		err = string_param_to_int(&resp_list, curr_index, &tmp, 16);
+		if (err) {
+			LOG_ERR("Could not parse cell_id, index %d, i %d error: %d",
+				curr_index, i, err);
+			goto clean_exit;
+		}
+
+		if (tmp > LTE_LC_CELL_EUTRAN_ID_MAX) {
+			tmp = LTE_LC_CELL_EUTRAN_ID_INVALID;
+		}
+		parsed_cell.id = tmp;
+
+		/* <plmn> */
+		len = sizeof(tmp_str);
+
+		curr_index++;
+		err = at_params_string_get(&resp_list, curr_index, tmp_str, &len);
+		if (err) {
+			LOG_ERR("Could not parse plmn, error: %d", err);
+			goto clean_exit;
+		}
+		tmp_str[len] = '\0';
+
+		/* Read MNC and store as integer. The MNC starts as the fourth character
+		 * in the string, following three characters long MCC.
+		 */
+		err = string_to_int(&tmp_str[3], 10, &parsed_cell.mnc);
+		if (err) {
+			LOG_ERR("string_to_int, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* Null-terminated MCC, read and store it. */
+		tmp_str[3] = '\0';
+
+		err = string_to_int(tmp_str, 10, &parsed_cell.mcc);
+		if (err) {
+			LOG_ERR("string_to_int, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <tac> */
+		curr_index++;
+		err = string_param_to_int(&resp_list, curr_index, &tmp, 16);
+		if (err) {
+			LOG_ERR("Could not parse tac in i %d, error: %d", i, err);
+			goto clean_exit;
+		}
+		parsed_cell.tac = tmp;
+
+		/* <ta> */
+		curr_index++;
+		err = at_params_int_get(&resp_list, curr_index,
+					&tmp);
+		if (err) {
+			LOG_ERR("Could not parse ta, error: %d", err);
```

Comment:
Suggestion: `ta` could be written as `timing advance` for clarity for those without knowledge of the parameter names in the AT reference. Same for other error messages

Reply:
changed